### PR TITLE
Add k8s auth api input

### PR DESCRIPTION
**Pull request by qemanuel**
### Added
- Input to define k8s authentication api for tf providers

### Fixed
- If condition for ingress-nginx serviceMonitor

### CODE REVIEW

File: helm.tf [hunk 1]
```diff
@@ -15,13 +15,15 @@ resource "helm_release" "ingress_nginx" {
   ]
 
   dynamic "set" {
+    for_each = var.helm_prometheus_enabled ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:18]: This isn't needed - if `var.helm_prometheus_enabled == false` then the helm value is overwritten with the default value (which is also false).

Behavior is the same, just one line less...

File: helm.tf [hunk 2]
```diff
@@ -15,13 +15,15 @@ resource "helm_release" "ingress_nginx" {
...
     content {
       name  = "controller.metrics.enabled"
       value = var.helm_prometheus_enabled
     }
   }
 
   dynamic "set" {
+    for_each = var.helm_prometheus_enabled ? ["do it"] : []
```
Comment (ivours) [inline helm.tf:26]: Same as https://github.com/nimbux911/terraform-aws-eks/pull/15/files#r915198440

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,10 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
-## [3.1.7] - 2022-07-06
+## [3.1.7] - 2022-07-04
+
+### Added
+- Input to define k8s authentication api for tf providers
 
 ### Fixed
-- Add default values for `controller.metrics.enabled` and `controller.metrics.serviceMonitor.enabled` to ingress-nginx Helm release in order to allow deploying it without having Prometheus as a dependency.
+- If condition for ingress-nginx serviceMonitor
```
[the remark this answers is not in the document]
Comment (qemanuel) [reply CHANGELOG.md]: You're right, i already removed that dependency which made no sense. Thanks

**Review by ivours — APPROVED** [review]

**Review by qemanuel — COMMENTED** [review]
Requested changes added

**Review by Ciejo — APPROVED** [review]

**Pull request merged**